lib/batch_reporter.ex

```diff
@@ -0,0 +1,88 @@
+defmodule TelemetryInfluxDB.BatchReporter do
+  @moduledoc """
+  This module handles batching events to be reported to Influx. It will call the provided
+  report function with any events that have been enqueued since the last report. If a batch_size
+  is not provided, the batch size will default to 1, which will send all events separately.
+  """
+
+  use GenServer
+
+  defmodule State do
+    defstruct [
+      :report_fn,
+      :batch_size,
+      :config,
```

Comment:
My understanding was that we didn't want `BatchReporter` to know anything about config; it's supposed to be a generic "batch reporter" module that can work with anything.  Instead, we should be enqueuing a payload like `{event, config}`, but BatchReporter doesn't know or care what that payload is.

Reply:
Yeah, we can do this instead, but tracking the config in the state felt pretty explicit, easy to test, and allowed us to quickly differentiate our batched events from their config. There were some thoughts about only updating the config in the state if it changes also.